common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -266,6 +266,21 @@ def get_items(self, course_key, **kwargs):
         return store.get_items(course_key, **kwargs)
 
     @strip_key
+    def get_course_summaries(self, **kwargs):
+        """
+        Returns a list containing the course information in CourseSummary objects.
+        Information contains `location`, `display_name`, `locator` of the courses in this modulestore.
+        """
+        course_summaries = {}
+        for store in self.modulestores:
+            for course_summary in store.get_course_summaries(**kwargs):
+                course_id = self._clean_locator_for_mapping(locator=course_summary.id)
+                if course_id not in course_summaries:
+                    # course is indeed unique. save it in result
```

Comment:
@nasthagiri I was just surprised to see that there were places where we "dealt" with duplicate course ids. I'm wondering if other things would be messed up if there were courses with the same id across multiple modulestores. So far, it looks like other areas, like [_get_modulestore_for_courselike](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/mixed.py#L176-L197), also assume that if there are duplicate course ids, take the first modulestore's course.

The thing I worry about is ending up in another orphan-type situation where we assume that course ids are totally unique, but also have a few protective measures in place in case they aren't. 

TL;DR: This is ok for here, but I'm going to raise this general issue to platform